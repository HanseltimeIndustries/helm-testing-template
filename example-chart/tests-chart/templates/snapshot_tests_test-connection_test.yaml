
{{- /*
# Helm unit testing
#
# TODO: parameterize this as needed by with values from your values.yaml
# 
# This folder is sorely lacking.  Please update anything that you change to
# ensure that we are proving any helm chart functionality that you've added.
# yaml-language-server: $schema=../helm-testsuite.json
*/}}
{{- range $idx, $env := $.Values.envs }}
suite: tests_test-connection snapshot {{ $env }} test
snapshotId: {{ $env }}
templates:
  - tests/test-connection.yaml
tests:
  - it: manifest should match snapshot
    set:
      env: {{ $env }}
    asserts:
    {{- /*
      If we have no documents snapshot doesn't handle that well
    */}}
    {{- if has "tests_test-connection" $.Values.noDocTemplates }}
      - hasDocuments:
        count: 0 
    {{- else }}
      - matchSnapshot: {}
    {{- end }}
---
{{- end }}
