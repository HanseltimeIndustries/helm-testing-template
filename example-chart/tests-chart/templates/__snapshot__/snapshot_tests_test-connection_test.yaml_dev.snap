manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: Pod
    metadata:
      annotations:
        helm.sh/hook: test
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-chart
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: example-chart-0.1.0
      name: RELEASE-NAME-example-chart-test-connection
    spec:
      containers:
        - args:
            - RELEASE-NAME-example-chart:80
          command:
            - wget
          image: busybox
          name: wget
      restartPolicy: Never
